docs/extensibility/create-extension.md

```diff
@@ -0,0 +1,392 @@
+          description: "Manage ConfigMap resources"
+          descriptionLong: "Use this command to manage ConfigMap resources."
+        subCommands:
+        - metadata:
+            name: create [flags]
+            description: "Create ConfigMap resource"
+            descriptionLong: "Use this command to create ConfigMap resource."
+          uses: resource_create
+          with:
+            resource:
+              apiVersion: v1
+              kind: ConfigMap
+              metadata:
+                name: cm-from-extension
+                namespace: default
+    ```
+
+4. Apply the new extension version and check if it works:
+
+    ```bash
+    kyma alpha configmap create
+    ```
+
+    You should see the following result:
+
+    ```bash
+    resource cm-from-extension applied
+    ```
+
+5. Extend the `create` command with resource-oriented features
+
+    Use [flags and args](./inputs.md#arguments-and-flags) to collect the defined name, namespace, and data from the user and pass them to the `resource_create` action using configuration under the `with` field using [Go templates](./inputs.md#go-templates) and the available [custom functions](./actions.md#custom-functions):
+
+    ```yaml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: my-extension
+      labels:
+        kyma-cli/extension: commands
+        kyma-cli/extension-version: v1
+    data:
+      kyma-commands.yaml: |-
+        metadata:
+          name: configmap [flags]
+          description: "Manage ConfigMap resources"
+          descriptionLong: "Use this command to manage ConfigMap resources."
+        subCommands:
+        - metadata:
+            name: create <resource_name> [flags]
+            description: "Create ConfigMap resource"
+            descriptionLong: "Use this command to create ConfigMap resource."
+          uses: resource_create
+          args:
+            type: string
+          flags:
+          - name: "namespace"
+            description: "ConfigMap namespace"
+            shorthand: "n"
+            type: string
+            default: "default"
+          - name: "from-literal"
+            description: "Data element in format <KEY>=<VALUE>"
+            type: map
+          with:
+            resource:
+              apiVersion: v1
+              kind: ConfigMap
+              metadata:
+                name: ${{ .args.value }}
+                namespace: ${{ .flags.namespace.value }}
+              data: ${{ .flags.fromliteral.value | toYaml }}
+    ```
+
+    > [!NOTE]
+    > In this case, we are building the `--from-literal` flag with the `map` type. With this, you can set this flag many times to collect more than one piece of data, but it requires additional conversion to an array using the [toYaml function](./actions.md#custom-functions). Also, the `.metadata.name` is updated because the command got new flags and args (following [quality standards](./README.md#extension-standards)).
```

Comment:
```suggestion
    > In this case, we are building the `--from-literal` flag with the `map` type. With this, you can set this flag many times to collect more than one piece of data, but it requires additional conversion to an array using the [toYaml function](./actions.md#custom-functions). Also, the `.metadata.name` is updated because the command got new flags and args, following [quality standards](./README.md#extension-standards).
```